react/features/toolbar/actions.web.js

```diff
@@ -0,0 +1,307 @@
+declare var interfaceConfig: Object;
+
+let toolbarTimeout = interfaceConfig.INITIAL_TOOLBAR_TIMEOUT;
+let toolbarTimeoutObject;
+
+/**
+ * Checks whether desktop sharing is enabled and whether
+ * we have params to start automatically sharing.
+ *
+ * @returns {Function}
+ */
+export function checkAutoEnableDesktopSharing(): Function {
+    return () => {
+        // XXX Should use dispatcher to toggle screensharing but screensharing
+        // hasn't been React-ified yet.
+
+        if (UIUtil.isButtonEnabled('desktop')
+                && config.autoEnableDesktopSharing) {
+            APP.UI.eventEmitter.emit(UIEvents.TOGGLE_SCREENSHARING);
+        }
+    };
+}
+
+/**
+ * Docks/undocks toolbar based on its parameter.
+ *
+ * @param {boolean} isDock - True if dock, false otherwise.
+ * @returns {Function}
+ */
+export function dockToolbar(isDock: boolean): Function {
+    return (dispatch: Dispatch<*>, getState: Function) => {
+        const state = getState();
+        const { visible } = state['features/toolbar'];
+
+        if (interfaceConfig.filmStripOnly) {
+            return;
+        }
+
+        if (isDock) {
+            // First make sure the toolbar is shown.
+            if (!visible) {
+                dispatch(showToolbar());
+            }
+
+            // Then clear the time out, to dock the toolbar.
+            clearTimeout(toolbarTimeoutObject);
+            toolbarTimeoutObject = null;
+        } else if (visible) {
+            toolbarTimeoutObject = setTimeout(() => {
+                dispatch(hideToolbar());
+            }, toolbarTimeout);
+        } else {
+            dispatch(showToolbar());
+        }
+    };
+}
+
+/**
+ * Hides the toolbar.
+ *
+ * @param {boolean} force - True to force the hiding of the toolbar without
+ * caring about the extended toolbar side panels.
+ * @returns {Function}
+ */
+export function hideToolbar(force: boolean = false): Function {
+    return (dispatch: Dispatch<*>, getState: Function) => {
+        const state = getState();
+        const {
+            alwaysVisible,
+            isHovered
+        } = state['features/toolbar'];
+
+        if (alwaysVisible) {
+            return;
+        }
+
+        clearTimeout(toolbarTimeoutObject);
+        toolbarTimeoutObject = null;
+
+        if (force !== true
+            && (isHovered
+            || APP.UI.isRingOverlayVisible()
+            || SideContainerToggler.isVisible())) {
+
+            toolbarTimeoutObject = setTimeout(() => {
+                dispatch(hideToolbar());
+            }, toolbarTimeout);
+        } else {
+            dispatch(setToolbarVisibility(false));
+            dispatch(setSubjectSlideIn(false));
+        }
+    };
+}
+
+/**
+ * Action that reset always visible toolbar to default state.
+ *
+ * @returns {Function}
+ */
+export function resetAlwaysVisibleToolbar(): Function {
+    return (dispatch: Dispatch<*>) => {
+        const alwaysVisible = config.alwaysVisibleToolbar === true;
```

Comment:
alwaysVisibleToolbar looks to be used in the jitsi-torture tests. I'm betting it's used to ensure the toolbar is clickable without worrying about hiding/showing.